### ISSUE TITLE
Hard-coded paadings lead to bugs when changing input image size in pytorch implementation

### ISSUE THREAD

**Issue by JohnTargaryen** [shepnerd/inpainting_gmcnn#56]
Greetings!
I change the input image size from 256,256,3 to 128,128,3, and mask size from 128,128 to 64,64 in the pytorch implementation.
However, there seem to be hard-coded padding in the network that causes bugs when changing input size.
![image](https://user-images.githubusercontent.com/17902928/76375722-80bca880-6381-11ea-948d-f2ac5ffd4fc0.png)

![image](https://user-images.githubusercontent.com/17902928/76375778-99c55980-6381-11ea-9571-458dd62ad3b4.png)

Could you please explain what these padding operation do? And how should I modify them to adapt to my desired input sizes?

**Comment by JohnTargaryen** [comment]
And I also wonder is multi-gpu training implemented in the pytorch implementation?

**Comment by shepnerd** [comment]
The padding size calculation using dilation conv is a bit different from the common conv, and you highlighted part just hard-codes these parameters. For the detailed rules, you can refer to this [post](https://discuss.pytorch.org/t/how-to-keep-the-shape-of-input-and-output-same-when-dilation-conv/14338).

If you intend to process 128x128 images, you can remove the fourth dilation conv layer (with dilation=16) and its corresponding padding operation, or ensure the dilation ratio will not exceed than a specific number (refer to the padding computation in that [post](https://discuss.pytorch.org/t/how-to-keep-the-shape-of-input-and-output-same-when-dilation-conv/14338)) and alter its corresponding padding.

For the multi-gpu training, you can use data parallel (torch.nn.DataParallel). As far as I know, current some implementations (maybe losses) are not supported in that and I will check that soon.
